plugins/docker/src/main/java/org/radargun/service/DockerService.java

```diff
@@ -0,0 +1,234 @@
+import org.radargun.config.Property;
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.Lifecycle;
+import org.radargun.traits.ProvidesTrait;
+import org.radargun.utils.EnvsConverter;
+import org.radargun.utils.Utils;
+
+/**
+ * A service that pulls a Docker image from a remote Docker registry and starts a container based
+ * on this image. A user can specify env variables that will be passed to the running container
+ * as well as exposed ports from the container.
+ * The container runs in "host" network mode and thus it has same network settings as the host. See
+ * http://www.dasblinkenlichten.com/docker-networking-101-host-mode/ for more information about
+ * host network mode.
+ *
+ * @author Martin Gencur
+ */
+@Service(doc = DockerService.SERVICE_DESCRIPTION)
+public class DockerService implements Lifecycle {
+
+   protected final Log log = LogFactory.getLog(getClass());
+   protected static final String SERVICE_DESCRIPTION = "Docker Service";
+   protected static final String DOCKER_HOST_NETWORK_MODE = "host";
+
+   private DockerCmdExecFactory dockerCmdExecFactory;
+   private DockerClient dockerClient;
+   private String dockerContainerId;
+   private boolean started = false;
+
+   @Property(doc = "Uri of docker server. Default is unix:///var/run/docker.sock")
+   protected String serverUri = "unix:///var/run/docker.sock";
+
+   @Property(doc = "Location of docker registry. Default is the official Docker registry.")
+   protected String dockerRegistry = "https://registry.hub.docker.com";
+
+   @Property(doc = "Username to be used to connect to Docker registry. Default is an empty value which can be used ONLY when the image is already available in the local Docker registry.")
+   protected String username;
+
+   @Property(doc = "Password to be used to connect to Docker registry. Default is an empty value which can be used ONLY when the image is already available in the local Docker registry.")
+   protected String password;
+
+   @Property(doc = "Image that will be downloaded from Docker registry and used to start a container", optional = false)
+   protected String image;
+
+   @Property(doc = "Name of the started Docker container. Empty by default.", optional = false)
+   protected String containerName;
+
+   @Property(doc = "Environment variables that will be passed to Docker container during startup. This is equivalent to starting Docker container with -e parameters. Empty by default.", converter = EnvsConverter.class)
+   protected Map<String, String> env = Collections.emptyMap();
+
+   @Property(doc = "The list of ports exposed by the container. This list should mimic ports exposed from a Docker image through EXPOSE. Empty by default.", converter = PortsConverter.class)
+   protected List<ExposedPort> exposedPorts = Collections.emptyList();
+
+   @ProvidesTrait
+   public Lifecycle getLifecycle() {
+      return this;
+   }
+
+   @Init
+   public void init() {
```

Comment:
OK. There's no indication that Init method should be used for this purpose. Why can't the logic be in the Init method? Is there any functional problem with this?